### ISSUE TITLE
Error on step 3 of installer

### ISSUE THREAD

**Issue by peimn** [statamic/v2-hub#1159]
### Expected behaviour
Error on step 3 of installer:
![screen shot 2016-12-13 at 10 09 07 pm](https://cloud.githubusercontent.com/assets/5695187/21153884/9ad6239a-c181-11e6-8f62-4abe5342f046.png)

### Server configuration
**Operating system**: macOS

**Web server:** NGINX

**PHP version:** 7.1

**Statamic version:** 2.1.18

**Comment by rrelmy** [comment]
It would be very helpful if you could attach the log file `local/storage/logs/statamic.log`

**Comment by peimn** [comment]
[2016-12-13 22:58:53] production.ERROR: ErrorException: Illegal string offset 'name' in /~PATH~/statamic/core/CP/Fieldset.php:240
Stack trace:
#0 /~PATH~/statamic/core/CP/Fieldset.php(240): Illuminate\Foundation\Bootstrap\HandleExceptions->handleError(2, 'Illegal string ...', '/private/var/ww...', 240, Array)
#1 /~PATH~/statamic/core/Http/Controllers/InstallerController.php(213): Statamic\CP\Fieldset->fieldtypes()
#2 /~PATH~/statamic/core/Http/Controllers/InstallerController.php(161): Statamic\Http\Controllers\InstallerController->processSettingsFields('system', Array, Object(Illuminate\Http\Request))
#3 [internal function]: Statamic\Http\Controllers\InstallerController->postSettings(Object(Illuminate\Http\Request))
#4 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/Controller.php(256): call_user_func_array(Array, Array)
#5 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(164): Illuminate\Routing\Controller->callAction('postSettings', Array)
#6 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(112): Illuminate\Routing\ControllerDispatcher->call(Object(Statamic\Http\Controllers\InstallerController), Object(Illuminate\Routing\Route), 'postSettings')
#7 [internal function]: Illuminate\Routing\ControllerDispatcher->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#8 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(139): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#9 /~PATH~/statamic/core/Http/Middleware/Installer.php(23): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#10 [internal function]: Statamic\Http\Middleware\Installer->handle(Object(Illuminate\Http\Request), Object(Closure))
#11 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#12 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#13 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#14 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(114): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#15 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/ControllerDispatcher.php(67): Illuminate\Routing\ControllerDispatcher->callWithinStack(Object(Statamic\Http\Controllers\InstallerController), Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'postSettings')
#16 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/Route.php(203): Illuminate\Routing\ControllerDispatcher->dispatch(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request), 'Statamic\\Http\\C...', 'postSettings')
#17 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/Route.php(134): Illuminate\Routing\Route->runWithCustomDispatcher(Object(Illuminate\Http\Request))
#18 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/Router.php(708): Illuminate\Routing\Route->run(Object(Illuminate\Http\Request))
#19 [internal function]: Illuminate\Routing\Router->Illuminate\Routing\{closure}(Object(Illuminate\Http\Request))
#20 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(139): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#21 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#22 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#23 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/Router.php(710): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#24 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/Router.php(673): Illuminate\Routing\Router->runRouteWithinStack(Object(Illuminate\Routing\Route), Object(Illuminate\Http\Request))
#25 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Routing/Router.php(635): Illuminate\Routing\Router->dispatchToRoute(Object(Illuminate\Http\Request))
#26 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(236): Illuminate\Routing\Router->dispatch(Object(Illuminate\Http\Request))
#27 [internal function]: Illuminate\Foundation\Http\Kernel->Illuminate\Foundation\Http\{closure}(Object(Illuminate\Http\Request))
#28 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(139): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#29 /~PATH~/statamic/core/Http/Middleware/PersistStache.php(22): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#30 [internal function]: Statamic\Http\Middleware\PersistStache->handle(Object(Illuminate\Http\Request), Object(Closure))
#31 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#32 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/VerifyCsrfToken.php(50): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#33 [internal function]: Illuminate\Foundation\Http\Middleware\VerifyCsrfToken->handle(Object(Illuminate\Http\Request), Object(Closure))
#34 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#35 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/View/Middleware/ShareErrorsFromSession.php(49): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#36 [internal function]: Illuminate\View\Middleware\ShareErrorsFromSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#37 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#38 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Session/Middleware/StartSession.php(62): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#39 [internal function]: Illuminate\Session\Middleware\StartSession->handle(Object(Illuminate\Http\Request), Object(Closure))
#40 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#41 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/AddQueuedCookiesToResponse.php(37): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#42 [internal function]: Illuminate\Cookie\Middleware\AddQueuedCookiesToResponse->handle(Object(Illuminate\Http\Request), Object(Closure))
#43 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#44 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Cookie/Middleware/EncryptCookies.php(59): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#45 [internal function]: Illuminate\Cookie\Middleware\EncryptCookies->handle(Object(Illuminate\Http\Request), Object(Closure))
#46 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#47 /~PATH~/statamic/core/Http/Middleware/Glide.php(43): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#48 [internal function]: Statamic\Http\Middleware\Glide->handle(Object(Illuminate\Http\Request), Object(Closure))
#49 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#50 /~PATH~/statamic/core/StaticCaching/Middleware/Retrieve.php(33): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#51 [internal function]: Statamic\StaticCaching\Middleware\Retrieve->handle(Object(Illuminate\Http\Request), Object(Closure))
#52 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#53 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Middleware/CheckForMaintenanceMode.php(44): Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#54 [internal function]: Illuminate\Foundation\Http\Middleware\CheckForMaintenanceMode->handle(Object(Illuminate\Http\Request), Object(Closure))
#55 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(124): call_user_func_array(Array, Array)
#56 [internal function]: Illuminate\Pipeline\Pipeline->Illuminate\Pipeline\{closure}(Object(Illuminate\Http\Request))
#57 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Pipeline/Pipeline.php(102): call_user_func(Object(Closure), Object(Illuminate\Http\Request))
#58 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(122): Illuminate\Pipeline\Pipeline->then(Object(Closure))
#59 /~PATH~/statamic/vendor/laravel/framework/src/Illuminate/Foundation/Http/Kernel.php(87): Illuminate\Foundation\Http\Kernel->sendRequestThroughRouter(Object(Illuminate\Http\Request))
#60 /~PATH~/index.php(155): Illuminate\Foundation\Http\Kernel->handle(Object(Illuminate\Http\Request))
#61 {main}

**Comment by jasonvarga** [comment]
Are you running Valet?

**Comment by edalzell** [comment]
2.1.8? Or 2.1.18?

**Comment by peimn** [comment]
@jasonvarga no, @edalzell it is 18 I've corrected it.

**Comment by rrelmy** [comment]
I have the exact same error on the settings page inside a installed statamic on a dev and live server but not locally with valet.

The server use a serverpilot setup with apache. I have different site with the same version on the same server without these error. I will look into it if I find some time

**Comment by cjf** [comment]
I think I have a fix...

Line 240 of /statamic/core/CP/Fieldset.php is `$config['name'] = $name;`

But the $config var may not be an array.  When I examine it, it is often just an empty string.

So my "fix" to get through the setup was to change the current line to...
`if ( ! is_array($config) && $config == '')
                $config = array();
$config['name'] = $name;`

This just initializes the $config value to an empty array in the event it is an empty string.  Without knowing what else is supposed to happen with that method, it may be unsafe, but it gets me through the install.

**Comment by rrelmy** [comment]
Actually this error happens only with PHP Version 7.1.
With PHP 7.0 and lower the empty string is silently converted to an array, PHP 7.1 throws a exception

**Comment by rrelmy** [comment]
The fix from cjf is incomplete

It is safer to check the fields when the fieldset is created.
Either in `API\Fieldset::fetch` or `CP\Fieldset->contents`

``` php
        if (!empty($contents['fields'])) {
            foreach ($contents['fields'] as $key => $config) {
                if (!is_array($config)) {
                    $contents['fields'][$key] = [];
                }
            }
        }
```

**Comment by SamAnthonyWP** [comment]
@rrelmy 

I am seeing the same issue. 

Is that snippet a replacement for line 240 in that file? Please excuse my lack of knowledge here. New to Statamic!

**Comment by rrelmy** [comment]
@SamAnthonyWP 

You can put this code before the line `$this->contents = $contents;` inside `statamic/core/CP/Fieldset.php` in the method `contents` (_around line 166_)

**Comment by SamAnthonyWP** [comment]
@rrelmy you rock. that worked 👍

**Comment by sheldonkotyk** [comment]
Note that this same error also arises when you go to a number of the settings pages in the CP. The 2.5 fix doesn't solve it.

**Comment by sheldonkotyk** [comment]
I just get the loading js circle.

**Comment by jasonvarga** [comment]
Its fixed in 2.5 since I've sent you a copy.

**Comment by sheldonkotyk** [comment]
ok perfect
